src/storage/v2/indices/vector_index.cpp

```diff
@@ -139,72 +74,69 @@ struct VectorIndex::Impl {
-      const auto dimension = index_spec[kDimension.data()].get<std::uint64_t>();
-      const auto size_limit = index_spec[kCapacity.data()].get<std::uint64_t>();
-      const auto resize_coefficient = index_spec.contains(kResizeCoefficient.data())
-                                          ? index_spec[kResizeCoefficient.data()].get<std::uint64_t>()
-                                          : kDefaultResizeCoefficient;
-
-      const auto label = LabelId::FromUint(name_id_mapper->NameToId(label_name));
-      const auto property = PropertyId::FromUint(name_id_mapper->NameToId(property_name));
-
-      result.emplace_back(
-          VectorIndexSpec{index_name, label, property, metric, scalar, dimension, size_limit, resize_coefficient});
+    CreateIndex(spec);
+    for (auto &vertex : vertices) {
+      UpdateVectorIndex(&vertex, LabelPropKey{spec->label, spec->property});
     }
   } catch (const std::exception &e) {
-    throw std::invalid_argument("Error parsing vector index spec: " + std::string(e.what()));
+    spdlog::error("Failed to create vector index {}: {}", spec->index_name, e.what());
+    return false;
   }
-
-  return result;
+  return true;
 }
 
-void VectorIndex::CreateIndex(const VectorIndexSpec &spec) {
-  const unum::usearch::metric_kind_t metric_kind = GetMetricKindFromConfig(spec.metric);
-  const unum::usearch::scalar_kind_t scalar_kind = GetScalarKindFromConfig(spec.scalar);
-
-  const unum::usearch::metric_punned_t metric(spec.dimension, metric_kind, scalar_kind);
-
-  // use the number of workers as the number of possible concurrent index operations
-  const unum::usearch::index_limits_t limits(spec.size_limit, FLAGS_bolt_num_workers);
+bool VectorIndex::DropIndex(std::string_view index_name) {
+  auto it = pimpl->index_name_to_label_prop_.find(index_name.data());
+  if (it == pimpl->index_name_to_label_prop_.end()) {
+    return false;
+  }
+  const auto &label_prop = it->second;
+  auto &[index, _, lock] = pimpl->index_.at(label_prop);
+  index.reset();
+  pimpl->index_.erase(label_prop);
+  pimpl->index_name_to_label_prop_.erase(it);
+  spdlog::info("Dropped vector index " + std::string(index_name));
+  return true;
+}
 
-  const auto label_prop = LabelPropKey{spec.label, spec.property};
-  pimpl->index_name_to_label_prop_.emplace(spec.index_name, label_prop);
-  pimpl->index_.emplace(label_prop, IndexItem{mg_vector_index_t::make(metric), spec, utils::RWSpinLock()});
-  if (pimpl->index_[label_prop].mg_index.try_reserve(limits)) {
-    spdlog::info("Created vector index " + spec.index_name);
-  } else {
-    throw std::invalid_argument("Failed to create vector index " + spec.index_name +
-                                " due to failed memory allocation. Try again with a smaller size limit.");
+void VectorIndex::Clear() {
+  pimpl->index_name_to_label_prop_.clear();
+  for (auto &[_, index_item] : pimpl->index_) {
+    index_item.mg_index.reset();
```

Comment:
I don't think you need individual reset...the map clear will destroy these values